### PR TITLE
zynqmp: updated to petalinux 2020.2

### PR DESCRIPTION
**Pull request by ierkiaga**
Build script updated to use Petalinux 2020.2 release. Custom recipes
removed in order to use upstream recipes insted for optee and python
devtool dependencies.

Signed-off-by: Ibai Erkiaga <ibai.erkiaga-elorza@xilinx.com>

### CODE REVIEW

**Comment by linnj** [comment]
Hi Joakim,  
We're still learning the system, so I apologize if this is a dumb question. I don't understand how a change to the zynqmp platform is affecting the HIKEY board running the test? Maybe I'm missing something obvious.
We have a new versal port that we will submit soon, but learning how this works is good.
Thanks
John

**Comment by jforissier** [comment]
@linnj I have seen the same HiKey crash 3 days ago when working on optee_os PR 4606 (the log has since been overwritten, the crash seems to happen randomly). So nothing related to this PR. I will try to reproduce the issue on Monday.

**Comment by jbech-linaro** [comment]
> on optee_os PR 4606 (the log has since been overwritten

Note, that by shorten the URL [1], IBART will filter out the old builds. So as long as you haven't pressed "restart", you should be able to find the old logs.
[1] https://optee.mooo.com:5000/logs/OP-TEE/optee_os/4606

**Comment by jforissier** [comment]
@jbech-linaro cool :) The link is https://optee.mooo.com:5000/logs/OP-TEE/optee_os/4606/641462945/07508bff9a1cc9f19a47e380999fc8fcfaf5ee58#end-01-hikey6220-test.

**Comment by ierkiaga** [comment]
@jforissier , did you had a chance to look into this? The automated stage build failure does not seem to be related to this pull request so not sure if this can be "safely" ignored.

**Comment by jforissier** [comment]
@ierkiaga sorry I have no clues about petalinux and zynqmp :-/ Perhaps @mimok who seems to be the original author of `zynqmp.mk` can comment?

**Comment by linnj** [comment]
I restarted the test manually yesterday and it succeeded.

**Comment by mimok** [comment]
Sorry for the late reply. Indeed there is no reason that this PR affects tests on HIKEY board. zynqmp build script was designed to be fully autonomous.

**Comment by jforissier** [comment]
@mimok the HiKey CI issue was fixed in https://github.com/OP-TEE/optee_os/pull/4628 although the review is still being finalized.
I was more asking for a review/acknowledgement of this PR. Sorry for being unclear.
Does this looks good to you? Any chance you could test it perhaps? Thanks.

**Comment by mimok** [comment]
@jforissier Ok, I'm going to review it and test it this week.

File: zynqmp.mk [hunk 1]
```diff
@@ -27,75 +26,63 @@ else
...
-.PHONY: check-petalinux
-check-petalinux:
+.PHONY: check
+check:
 ifndef PETALINUX_VER
 	$(error You have to source Petalinux settings)
 endif
-ifneq ($(PETALINUX_VER),2018.2)
+ifneq ($(PETALINUX_VER),2020.2)
 	$(error This makefile only support Petalinux 2018.2)
```
Comment (mimok) [inline zynqmp.mk:37]: Petalinux version should be updated in the error message

File: zynqmp.mk [hunk 2]
```diff
@@ -27,75 +26,63 @@ else
...
-	@cp -r $(PETALINUX_CFG_PATH)/python/* \
-	    $(PRJ_PATH)/project-spec/meta-user/recipes-devtools/python/
-	
-petalinux-config: check-petalinux
-	$(call set_optee_version,$(OPTEE_VER),$(PRJ_PATH)/project-spec/meta-user/recipes-apps/optee-test/optee-test.bbappend)
-	$(call set_optee_version,$(OPTEE_VER),$(PRJ_PATH)/project-spec/meta-user/recipes-apps/optee-client/optee-client.bbappend)
-	$(call set_optee_version,$(OPTEE_VER),$(PRJ_PATH)/project-spec/meta-user/recipes-bsp/optee-os/optee-os.bbappend)
-	@petalinux-config -p $(PRJ_PATH) --oldconfig
+create: check
+	# Create TMP directory to avoid issues with .. in the path name
```
Comment (mimok) [inline zynqmp.mk:41]: Comments should be prefixed with @ to prevent displaying them on terminal

**Comment by mimok** [comment]
I have tested the PR using ZCU102 BSP and Qemu and everything seem OK.

However there is two failure during xtest execution
`regression_1027 FAILED first error at /usr/src/debug/optee-test/3.11.0-r0/git/host/xtest/regression_1000.c:2123
regression_1028 FAILED first error at `/usr/src/debug/optee-test/3.11.0-r0/git/host/xtest/regression_1000.c:2188`
Not sure if this failure are really related to this PR.

I cannot test the generated package on Ultra96-v1 as this board is no more supported now but i think test on ZCU102 and Qemu is sufficient.

Last thing, documentation should be updated accordingly as there is some minor differences with the original implementation.

File: zynqmp.mk (hunk 1 above)
Comment (ierkiaga) [reply zynqmp.mk:37]: Right, let me address it in the next iteration

**Comment by ierkiaga** [comment]
The xtest 1027 and 1028 errors are expected [the required kernel patch](https://github.com/OP-TEE/optee_test/blob/master/host/xtest/regression_1000.c#L2079) to run those tests are not present in the kernel release used in Petalinux 2020.2.

**Comment by jforissier** [comment]
@ierkiaga thanks for the update.

@mimok would you like to give some formal `Tested-by:` and `Acked-by:` tags for this? No obligation, but it would be helpful.

**Comment by mimok** [comment]
> @mimok would you like to give some formal `Tested-by:` and `Acked-by:` tags for this? No obligation, but it would be helpful.

@jforissier, No problem but not sure how to add these tags. Should I leave a comment with these tags?

**Comment by jforissier** [comment]
> > @mimok would you like to give some formal `Tested-by:` and `Acked-by:` tags for this? No obligation, but it would be helpful.
> 
> @jforissier, No problem but not sure how to add these tags. Should I leave a comment with these tags?

Yes, exactly. Then the patch author is supposed to add them to the commit description and force push.

**Comment by mimok** [comment]
Acked-by: Michael Grand <michael.grand.mg@gmail.com>
Tested-by: Michael Grand <michael.grand.mg@gmail.com>

**Comment by ierkiaga** [comment]
Thanks @mimok for the review.
@jforissier is there anything else I need to do?

**Comment by jforissier** [comment]
@ierkiaga, no everything is good. The IBART error is caused by your branch being a bit too old wrt. master, but let's ignore that, the commit will get rebased automatically when I merge this. Thanks all!

**Comment by ierkiaga** [comment]
I have notice that the rest of the targets are using pure git repositories and buildroot for generating the images. Is it the preferred way? I guess that it would help more people being able to test/review code (as petalinux installation would not be required) and a better integration with CI infrastructure. Any comments about it?

**Comment by mimok** [comment]
In my opinion, too many work is needed to switch to build root while Xilinx already did it trough petalinux. In addition, I think that many Xilinx customers already use Petalinux. This is why I have chosen to keep petalinux even if this is not the preferred way. Maybe you could follow the standard OP-TEE build process if you have the right skills (which was not my case when I started this work) and enough time.

**Pull request merged**